### PR TITLE
added support for devenv

### PR DESCRIPTION
**Pull request by DGollings**
made this for easy dependency management for myself, but you might be interested. This sets up an environment automatically, just type `direnv allow` and it'll download Python, do all the venv and pip stuff and be ready for a simple `python src/extract_otp_secrets.py ./file.txt`

### CODE REVIEW

File: devenv.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+inputs:
```
Comment (scito) [inline devenv.yaml:1]: Can you enhance the README.md and add the usage of devenv?
There is a section about dev environments.

File: devenv.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  nixpkgs:
+    url: github:cachix/devenv-nixpkgs/rolling
```
Comment (scito) [inline devenv.yaml:3]: What is this URL for?

**Review by scito — COMMENTED** [review]
Thanks for your contribution!
I've added some questions.

File: .envrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+source_url "https://raw.githubusercontent.com/cachix/devenv/95f329d49a8a5289d31e0982652f7058a189bfca/direnvrc" "sha256-d+8cBpDfDBj41inrADaJt+bDWhOktwslgoP5YiGJ1v0="
```
[the remark this answers is not in the document]
Comment (DGollings) [reply .envrc:1]: its simply configured that way by running `devenv init`

File: devenv.yaml (hunk 2 above)
Comment (DGollings) [reply devenv.yaml:3]: its simply configured that way by running `devenv init`

**Comment by DGollings** [comment]
Done :)

**Comment by scito** [comment]
Thanks for your contribution!

**Pull request merged**